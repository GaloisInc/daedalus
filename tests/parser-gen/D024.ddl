-- test complement of class character

def A =
  First
    block $$ = $[0xFF] ; $[0x00]
    $[!0xFF]


def Main =
  block
    A
