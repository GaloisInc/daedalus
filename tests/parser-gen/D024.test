D024.ddl
--dump-gen