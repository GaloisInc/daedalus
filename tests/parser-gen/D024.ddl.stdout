module D024
 
--- Imports:
 
--- Type defs:
 
--- Rules:
 
D024.A : Grammar (uint 8) =
  Choose biased
    { do ($$ : uint 8) <- Match {0xFF}
         @Match {0x00}
         pure $$
    | Match (!{0xFF})
    }
 
D024.Main : Grammar (uint 8) =
  do ($$ : uint 8) <- D024.A
     pure $$
 
D024.A_4 : Grammar {} =
  Choose biased
    { do @Match {0xFF}
         @Match {0x00}
    | @Match (!{0xFF})
    }
 
D024.Main_5 : Grammar {} =
  D024.A_4
